File: backend/app/core/upgrades/curses/fragility_2.py
from ..upgrade_implementation import UpgradeImplementation, UpgradeAppliers


class Fragility2(UpgradeImplementation):
    id = 'fragility_2'
    name = 'Fragility 2'
    description = 'Increased damage taken by 3.5%'
    rarity = 'epic'
    type = 'effect'
    cost = 0

    def can_apply(self, state: dict) -> bool:
        return True

    def apply(self, appliers: UpgradeAppliers) -> None:
        appliers.effects.register_effect('protection', {'value': 0.035})

    def validate(self, recorded_stats: dict) -> bool:
        return True

File: frontend/src/game/upgrades/curses/fragility_2.ts
import { Upgrade, type UpgradeDef, type DamageRef } from '../Upgrade'
import { RarityID, UpgradeTypeID } from '../../data/ID'

export const Fragility2Def: UpgradeDef = {
  id: "fragility_2",
  name: "Fragility 2",
  description: "Increased damage taken by 3.5%",
  rarity: RarityID.Epic,
  upgradeType: UpgradeTypeID.Effect,
  cost: 0,
  value: 0.035,
  isMultiplier: true,
  effect: "protection",
  stackable: true,
  maxStacks: 1,
  curse: true,
}

export class Fragility2 extends Upgrade {
  onApply(): void {}

  modifyPlayerHurt(damage: DamageRef): void {
    damage.amount *= 1 + this.def.value!
  }
}
